File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        int ans=0;
        Stack<Integer> stack = new Stack<>();
        for(int i=0;i<tokens.length;i++){
            String s = tokens[i];
            if(s.equals("+") || s.equals("-")||s.equals("*")||s.equals("/")){
                int a = stack.peek();
                stack.pop();
                int b= stack.peek();
                stack.pop();
                if(s.equals("+")) stack.push(b+a);
                else if(s.equals("-")) stack.push(b-a);
                else if(s.equals("*")) stack.push(b*a);
                else stack.push(b/a);
            }else stack.push(Integer.parseInt(s));
        }
        return stack.peek();
        
    }
}

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int>stack;
        for(string s:tokens){
            if(s=="*"){
                int n1=stack.top();
                stack.pop();
                int n2=stack.top();
                stack.pop();

                stack.push(n1*n2);
            }else if(s=="-"){
                int n1=stack.top();
                stack.pop();
                int n2=stack.top();
                stack.pop();

                stack.push(n2-n1);
            }else if(s=="+"){
                int n1=stack.top();
                stack.pop();
                int n2=stack.top();
                stack.pop();

                stack.push(n1+n2);
            }else if(s=="/"){
                int n1=stack.top();
                stack.pop();
                int n2=stack.top();
                stack.pop();

                stack.push(n2/n1);
            }else{
                stack.push(stoi(s));
            }
        }
        return stack.top();
    }
};